### ISSUE TITLE
Cell segmentation results not match images well.

### ISSUE THREAD

**Issue by windygoo** [mahmoodlab/HEST#82]
I intend to display cell segmentation results on a patch image with the following code, however, I found the segmentation results not match images well. How to address this issue?

`

    import h5py
    with h5py.File('patches/INT1.h5', 'r') as f:
        coord = f['coords'][0]
        img = f['img'][0]

    cell_seg = json.load(open('cellvit_seg/INT1_cellvit_seg.geojson'))
    
    all_bboxs = []
    for res in cell_seg:
        cls_bboxs = []
        for coords in res["geometry"]["coordinates"]:
            coords = coords[0]
            x1, y1 = np.min(coords, axis=0)
            x2, y2 = np.max(coords, axis=0)
            cls_bboxs.append([x1, y1, x2, y2])
        all_bboxs.append(cls_bboxs)
    
    import torch
    from torchvision.ops import box_iou

    all_bboxs = torch.from_numpy(np.concatenate(all_bboxs, axis=0))

    start_x, end_x = coord[0], coord[0] + 224
    start_y, end_y = coord[1], coord[1] + 224

    region_limits = [[start_x, start_y, end_x, end_y]]

    region_limits = torch.tensor(region_limits)
    
    ious = box_iou(region_limits, all_bboxs)
    
    idxs = torch.where(ious > 0)[1]

    all_contours = []
    for c, res in enumerate(cell_seg):
        cls_bboxs = []
        for coords in res["geometry"]["coordinates"]:
            all_contours.append(coords)

    import cv2

    inst_map = np.zeros((224, 224), dtype=int)
    for inst_id, idx in enumerate(idxs):
        contour = np.array(all_contours[idx])[0]
        
        contour[:, 0] = np.clip(contour[:, 0], start_x, end_x)
        contour[:, 1] = np.clip(contour[:, 1], start_y, end_y)

        if len(contour) < 3:
            continue
        
        contour = contour - np.array([[start_x, start_y]])

        
        contour_points = contour.reshape(-1, 1, 2).astype(np.int32)
        temp_mask = np.zeros((224, 224), dtype=np.uint8)
        cv2.fillPoly(temp_mask, [contour_points], 1)
        
        cv2.drawContours(img, [contour_points], -1, (0, 255, 0), 2)

    io.imsave('test.png', img)

`

![test](https://github.com/user-attachments/assets/da64b935-0dd5-482e-8534-e4c562b6afe5)

**Comment by pauldoucet** [comment]
Hi @windygoo,

We extracted patches of 224x224 pixels at 0.5 um/px, which means that we rescaled the patches to 0.5 um/px before saving them. The .geojson files are saved at source resolution. You need to rescale the cell coordinates by the downsample factor. You can extract it as follows:

```
with h5py.File('INT1.h5', 'r') as f:
    downsample = dict(f['img'].attrs)['downsample']
```

**Comment by windygoo** [comment]
> Hi @windygoo,
> 
> We extracted patches of 224x224 pixels at 0.5 um/px, which means that we rescaled the patches to 0.5 um/px before saving them. The .geojson files are saved at source resolution. You need to rescale the cell coordinates by the downsample factor. You can extract it as follows:
> 
> ```
> with h5py.File('INT1.h5', 'r') as f:
>     downsample = dict(f['img'].attrs)['downsample']
> ```

Thanks, I have fixed this problem.

**Comment by windygoo** [comment]
{'factor': 2.268842666666667, 'patch_size': 224, 'pixel_size': 0.22037667368737773}
ZEN80.h5
{'factor': 2.2644643719653685, 'patch_size': 224, 'pixel_size': 0.22080276739617732}
ZEN81.h5
{'factor': 2.2644472065322874, 'patch_size': 224, 'pixel_size': 0.2208044411711794}
ZEN79.h5


These three files do not have 'downsample' attribute. Run code {downsample = dict(f['img'].attrs)['downsample']} could raise KeyError.

**Comment by JosephDiPalma** [comment]
@windygoo Would you be willing to share the updated code with the downsampling?
